### PR TITLE
Successfully converted the MediaRecorder to an Audio Recorder

### PR DESCRIPTION
**Pull request by christopherWilliams98**
As Proposed by colleague Baptiste, it is beneficial to use an AudioRecorder for the microphone rather than a MediaRecorder. The reason being that an audio recorder will use a buffer which can be sampled every few miliseconds in order to do the frequency analysis in real time, in addition, the audio samples are cleaner as they are in PCM format which is a direct digital representation of analog signals, instead of the previous AAc format we used which was compressed and impure.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <option name="noGCSettingWarning" value="true" />
   </component>
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_17" default="true" project-jdk-name="Android Studio default JDK" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="true" project-jdk-name="Android Studio default JDK" project-jdk-type="JavaSDK">
```
Comment (Netzu00) [inline .idea/misc.xml:7]: Why did you switch this?

File: app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt [hunk 2]
```diff
@@ -103,59 +117,65 @@ class RecFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
...
+
+    // Method to stop recording
     private fun stopRecording() {
-        //stop recording
-        mediaRecorder.stop()
-        //release the media recorder
-        mediaRecorder.release()
-        //set recording not in progress
+        // Stop recording
+        audioRecorder.stop()
```
Comment (Netzu00) [inline app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt:169]: Does this have a way to fail? And in that case could still be locking the file? If not, ignore this comment

File: app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt [hunk 3]
```diff
@@ -103,59 +117,65 @@ class RecFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
...
+        Thread(Runnable {
+            while (isRecording) {
+                val bytesRead = audioRecorder.read(buffer, 0, bufferSize)
+                if (bytesRead > 0) {
+                    try {
+                        outputStream.write(buffer, 0, bytesRead)
+                    } catch (e: IOException) {
+                        e.printStackTrace()
+                    }
                 }
```
Comment (Netzu00) [inline app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt:147]: if(bytesRead <= 0), should that be an error? Could add a toast as a small indication?

**Review by Netzu00 — APPROVED** [review]
Really good! See comments for small change suggestions

File: app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt (hunk 2 above)
Comment (christopherWilliams98) [reply app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt:133]: I don't believe this has a way to fail and lock the file, as far as I know.

File: .idea/misc.xml (hunk 1 above)
Comment (christopherWilliams98) [reply .idea/misc.xml]: Thank you for pointing this out, this must have been done by my android studio sneakily. I will revert this change.

Comment (christopherWilliams98) [reply .idea/misc.xml]: I have now reverted this

**Pull request merged**